x/precisebank/README.md

```diff
@@ -127,50 +126,92 @@ $$a'(1) = a(1) - a$$
 
 $$a'(2) = a(2) + a$$
 
-The reserve account and remainder must also be updated to reflect the change in
-the total supply of fractional units
+The reserve account must also be updated to reflect the change in the total
+supply of fractional units.
 
 $$b(R) \cdot C = \sum_{n \in \mathcal{A}}{f(n)} + r$$
 
 $$b'(R) \cdot C = \sum_{n \in \mathcal{A}}{f'(n)} + r'$$
 
+With these two formulas, we can determine the new remainder and reserve by using
+the delta of the sum of fractional units and the remainder.
+
 $$(b'(R)-b(R)) \cdot C = \sum_{n \in \mathcal{A}}{f'(n)} - \sum_{n \in \mathcal{A}}{f(n)} + r' - r$$
 
+Since only two accounts are involved in the transfer, we can use the two account
+balances in place of the fractional sum delta.
+
 $$(b'(R)-b(R)) \cdot C = f'(1) - f(1) + f'(2) - f(2) + r' - r$$
 
 #### Remainder does not change
 
-$$(b'(R)-b(R)) \cdot C = f'(1) - f(1) + f'(2) - f(2) + r' - r \mod{C}$$
+Take $\mod{C}$ of both sides of the equation.
+
+$$(b'(R)-b(R)) \cdot C \mod{C} = [f'(1) - f(1) + f'(2) - f(2) + r' - r] \mod{C}$$
+
+Since $C$ is a multiple of $C$, the left side of the equation is $0$.
+
+$$0 = f'(1) - f(1) + f'(2) - f(2) + r' - r \mod{C}$$
+
+Replace $f'(1)$ and $f'(2)$ with their definitions in terms of $f(1)$ and $f(2)$.
 
 $$0 = (f(1) - a)\bmod{C} - f(1) + (f(2) + a)\bmod{C} - f(2) + r' - r \mod{C}$$
 
+This can be simplified to:
+
 $$0 = f(1) - a - f(1) + f(2) + a - f(2) + r' - r \mod{C}$$
 
+Canceling out terms $a$, $f(1)$ and $f(2)$.
+
 $$0 = r' - r \mod{C}$$
 
+By the quotient remainder theorem, we can express $r' - r$ as:
+
 $$q * C = r' - r$$
 
+for some integer $q$.
+
+With our known range of $r$ and $r'$:
+
 $$0 \leq r' < C, 0 \leq r < C$$
 
+We can see that $r' - r$ must be in the range
+
 $$ -C < r' - r < C$$
 
+This implies that $q$ must be $0$ as there is no other integer $q$ that satisfies the inequality.
+
+$$ -C < q * C < C$$
+
 $$q = 0$$
 
 $$ r' - r = 0$$
 
+Therefore, the remainder does not change during a transfer.
+
 #### Reserve
 
+The reserve account must be updated to reflect the change in the total supply of fractional units.
+
+The change in reserve is determined by the change in the fractional units of the two accounts.
+
 $$(b'(R)-b(R)) \cdot C = f'(1) - f(1) + f'(2) - f(2)$$
 
+For $f'(1)$, we can represent the new fractional balance as:
+
 $$f'(1) = f(1) - a \mod{C}$$
 
 $$f'(1)\bmod{C}= f(1)\bmod{C} - a \bmod{C} \mod{C}$$
 
 $$f'(1) = f(1) - a \bmod{C} \mod{C}$$
 
+This results in two cases for $f'(1)$:
+
 $$f'(1) = \begin{cases} f(1) - a\bmod{C} & f'(1) \leq f(1) \\
```

Comment:
It's probably more clear and formal to write

$$f'(1) = \begin{cases} f(1) - a\bmod{C} & 0 \leq f(1) -a\bmod{C} \\
f(1) - a\bmod{C} + C & 0 > f(1) - a\bmod{C} \end{cases}$$

at this step as we shouldn't include the result as the definition. This was lazy typing on my part.

Checking $f'(1)$ and $f(1)$ in later steps is an optimization since the assumption is $f'(1)$ and $f(1)$ are already known and noticing

$$f'(1) \leq f(1) \Longleftrightarrow  f'(1) = f(1) - a\bmod{C} $$

$$f'(1) > f(1) \Longleftrightarrow  f'(1) = f(1) - a\bmod{C} + C$$